src/shared_realm.cpp

```diff
@@ -657,112 +656,87 @@ util::Optional<int> Realm::file_format_upgraded_from_version() const
     return util::none;
 }
 
-Realm::HandoverPackage::HandoverPackage(HandoverPackage&&) = default;
-Realm::HandoverPackage& Realm::HandoverPackage::operator=(HandoverPackage&&) = default;
-
-// Precondition: `m_version` is not greater than `new_version`
-// Postcondition: `m_version` is equal to `new_version`
-void Realm::HandoverPackage::advance_to_version(VersionID new_version)
+template <typename T>
+realm::ThreadSafeReference<T> Realm::obtain_thread_safe_reference(T value)
```

Comment:
I think we should take `value` by const reference here to avoid creating an unnecessary copy.